### PR TITLE
Add primary category type

### PR DESCRIPTION
**Pull request by pmarius22**
Resolves https://github.com/odpi/egeria/issues/4838

### CODE REVIEW

File: open-metadata-publication/website/open-metadata-types/0335-Primary-Category.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!-- SPDX-License-Identifier: CC-BY-4.0 -->
+<!-- Copyright Contributors to the ODPi Egeria project. -->
+
+# 0335 Primary Category 
+
+This Classification marks the Primary Category associated with a [GlossaryTerm](0330-Terms.md) and helps to identify it in the lineage graph. 
+It contains a single filed, `categoryQualifiedName` which represents the qualified name of the primary category associated with the term.
```
Comment (mandy-chessell) [inline open-metadata-publication/website/open-metadata-types/0335-Primary-Category.md:7]: Spelling mistake in filed - should be property.

Comment (mandy-chessell) [inline open-metadata-publication/website/open-metadata-types/0335-Primary-Category.md:7]: Add link to category

File: open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive.java [hunk 2]
```diff
@@ -145,7 +145,7 @@ public OpenMetadataArchive getOpenMetadataArchive()
      */
     public void getOriginalTypes()
     {
-        OpenMetadataTypesArchive2_7  previousTypes = new OpenMetadataTypesArchive2_7(archiveBuilder);
+        OpenMetadataTypesArchive2_9  previousTypes = new OpenMetadataTypesArchive2_9(archiveBuilder);
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive.java:148]: should be 2_8

File: open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive2_9.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive2_9.java:1]: The new type definition should go in file OpenMetadataTypesArchive.java

File: open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive2_9.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+     */
+    private void add0335PrimaryCategoryClassification()
+    {
+        this.archiveBuilder.addClassificationDef(addPrimaryCategoryClassification());
+    }
+
+    private ClassificationDef addPrimaryCategoryClassification()
+    {
+        final String guid            = "3a6c4ba7-3cc5-48cd-8952-bwra92da016d";
+        final String name            = "PrimaryCategoryClassification";
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive2_9.java:176]: Do not need the word Classification in the title - it should be "PrimaryClassification"

**Review by mandy-chessell — COMMENTED** [review]
The images should have a border of 5, and shadow and transparency set,

**Review by mandy-chessell — CHANGES_REQUESTED** [review]
The new type definition needs to be moved to OpenMetadataTypesArchive.java before merging

**Comment by mandy-chessell** [comment]
The image for https://github.com/odpi/egeria/blob/ed0f2a81dd5cf736fe631d9e54c08bfcf507ca7d/open-metadata-publication/website/open-metadata-types/0335-Primary-Category.png

* has a spelling mistake in - PrimaryCategory
* Has the wrong sized box around it - it should be the same size as the other pages.

**Review by lpalashevski — APPROVED** [review]
looks good. @mandy-chessell can you please take a look and approve?

**Review by cmgrote — APPROVED** [review]

File: open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive.java [hunk 5]
```diff
@@ -415,5 +416,52 @@ private TypeDefPatch updatePortEntity()
...
+         */
+        List<TypeDefAttribute> properties = new ArrayList<>();
+        TypeDefAttribute       property;
+
+        final String attribute1Name            = "categoryQualifiedName";
+        final String attribute1Description     = "The qualified name of the primary category of a GlossaryTerm.";
+        final String attribute1DescriptionGUID = null;
+
+        property = archiveHelper.getStringTypeDefAttribute(attribute1Name,
+                attribute1Description,
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-archives/open-metadata-types/src/main/java/org/odpi/openmetadata/opentypes/OpenMetadataTypesArchive.java:456]: Code layout issue - need to align parametersunder attribute1Name

**Review by mandy-chessell — COMMENTED** [review]
The area-3-glossary-overview.png is missing the shadow.

**Review by mandy-chessell — APPROVED** [review]

**Comment by mandy-chessell** [comment]
There are a coupld of minor issues still to resolve - however can be done in a separate PR

**Pull request merged**